corehq/apps/hqwebapp/static/hqwebapp/spec/email_validator_spec.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-env mocha */
+hqDefine("hqwebapp/spec/email_validator_spec", [
+    'hqwebapp/js/constants',
+], function (
+    constants,
+) {
+    describe('email_validator', function () {
+        const re = constants.EMAIL_VALIDATION_REGEX;
+        const testEmail = email => re.test(email);
+
+        it('should allow simple email addresses', function () {
+            assert.ok(testEmail('simple@example.com'));
+        });
+
+        it('should allow capital letters in the local part', function () {
+            assert.ok(testEmail('Capital@example.com'));
+            assert.ok(testEmail('VERYCAPITAL@example.com'));
+        });
+
+        it('should allow capital letters in the domain part', function () {
+            assert.ok(testEmail('email@Example.com'));
+            assert.ok(testEmail('email@EXAMPLE.COM'));
+        });
+
+        it('should allow digits in the local part', function () {
+            assert.ok(testEmail('email123@example.com'));
+        });
+
+        it('should allow specified special characters in the local part', function () {
+            assert.ok(testEmail('.\x01!#$%&"\'*+/=?^_`{|}~-@example.com'));
```

Comment:
wild

Reply:
Agreed! I think this is one of the places where, strictly speaking, some of these characters should only be allowed based on context rather than all the time, e.g. a quote should always be paired. But the existing expression doesn't test for that.